3rdParty/fuerte/src/H2Connection.cpp

```diff
@@ -373,24 +370,29 @@ void H2Connection<T>::finishConnect() {
   ssize_t rv = nghttp2_session_upgrade2(_session, (uint8_t const*)packed.data(),
                                         packed.size(), /*head*/ 0, nullptr);
   if (rv < 0) {
-    this->shutdownConnection(Error::ProtocolError);
+    FUERTE_ASSERT(false);
+    this->shutdownConnection(Error::ProtocolError, "error during upgrade");
     return;
   }
+  
+  // simon: important otherwise big responses fail
+  // increase connection window size up to window_size
+  rv = nghttp2_session_set_local_window_size(_session, NGHTTP2_FLAG_NONE, 0, window_size);
+  FUERTE_ASSERT(rv == 0);
 
   auto req = std::make_shared<std::string>();
   req->append("GET / HTTP/1.1\r\nConnection: Upgrade, HTTP2-Settings\r\n");
   req->append("Upgrade: h2c\r\nHTTP2-Settings: ");
   req->append(encoded);
   req->append("\r\n\r\n");
 
-  std::cout << "sending request '" << *req << "'\n";
-
   asio_ns::async_write(
       this->_proto.socket, asio_ns::buffer(req->data(), req->size()),
       [self(Connection::shared_from_this())](auto const& ec,
                                              std::size_t nsend) {
         auto& me = static_cast<H2Connection<T>&>(*self);
         if (ec) {
+          FUERTE_ASSERT(false);
```

Comment:
we should remove this. Isn't it expected that errors can happen during async_write?